### PR TITLE
Add insert-koin along with mockito

### PR DESCRIPTION
**Pull request by david-schroeter**
#### We did this PR, @2Tricky4u  and me, together.

In this PR we did the following things:
- add `insert-koin` dependency injection system
- add dependency to use `mockito`
- modify `AdviceFragmentTest` to use this new injection system along with `mockito` to easily mock the `FirebaseDB` instance, and have the result of call to db instantaneously, and remove the overhead of having to work with a `CompletableDeferred`.

### CODE REVIEW

**Review by LucaCarroz — APPROVED** [review]
That's a very nice way of injecting a new database implementation to test our code, thanks a lot! That will very likely be useful for the following sprints!
No comments on the code, everything looks good :)

File: app/src/main/java/com/github/sdp_begreen/begreen/firebase/DB.kt [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    /**
+     * Retrieves the set of all bin locations currently present in the database
+     *
+     * @return the set of locations of all the bins, or null if an error happened
+     */
+    suspend fun getAllBinLocations(): Set<LatLng>?
+
+    /**
+     * Retrieves the list of advices from the realtime database
```
Comment (SolalPirelli) [inline app/src/main/java/com/github/sdp_begreen/begreen/firebase/DB.kt:133]: terminology: in English, "advice" is uncountable, so you have "a piece of advice" but not "an advice". Perhaps "tips" is a better word for this.

File: app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt [hunk 2]
```diff
@@ -249,14 +168,47 @@ object FirebaseDB {
...
+        // Looks like manually deleting the unchecked cast is the only way to remove the warnings
+        @Suppress("UNCHECKED_CAST")
+        val childrens = databaseReference.child(BIN_LOCATION_PATH).get().await().value
+                as? Map<String, Any> ?: return null
+
+        val locations = mutableSetOf<LatLng>()
+        for (child in childrens.values) {
+            // Converts the Map back to a LatLng. If some child does not have the correct format, skip it
+            @Suppress("UNCHECKED_CAST")
+            val location = child as? HashMap<String, Double>? ?: child as? HashMap<String, Long>? ?: continue
```
Comment (SolalPirelli) [inline app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt:187]: this is really weird from a design perspective

File: app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/AdviceFragmentTest.kt [hunk 3]
```diff
@@ -7,42 +7,41 @@ import androidx.test.espresso.matcher.ViewMatchers.*
...
-                Firebase.storage.useEmulator("10.0.2.2", 9199)
-                Firebase.auth.useEmulator("10.0.2.2", 9099)
-            } catch (_:java.lang.IllegalStateException){}
-        }
-    }
+
+    private val db: DB = mock(DB::class.java)
+
+    @get:Rule
+    val koinTEstRule = KoinTestRule(
```
Comment (AlSchlo) [inline app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/AdviceFragmentTest.kt:29]: Little typo

File: app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt [hunk 4]
```diff
@@ -143,16 +108,7 @@ object FirebaseDB {
...
-     *
-     * @param image the profile picture to store
-     * @param userId the [User] for whom to add the pictures
-     * @return the updated [PhotoMetadata] with the current id of the image we stored
-     *
-     * @throws IllegalArgumentException if the [userId] was blank or empty
-     */
-    suspend fun storeUserProfilePicture(image: Bitmap, userId: String, metadata: PhotoMetadata): PhotoMetadata? {
+    override suspend fun storeUserProfilePicture(image: Bitmap, userId: String, metadata: PhotoMetadata): PhotoMetadata? {
         if (userId.isBlank())
```
Comment (AlSchlo) [inline app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt:112]: A lot of code duplication here

File: app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt (hunk 2 above)
Comment (david-schroeter) [reply app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt:187]: Indeed I totaly agree with you, we will try to improve that.

File: app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt (hunk 4 above)
Comment (david-schroeter) [reply app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt:112]: I am sorry, but I do not really see about what duplication you are actually speaking of ?

**Pull request merged**